pkg/kubelet/server/server.go

```diff
@@ -173,9 +183,14 @@ func ListenAndServeKubeletServer(
 }
 
 // ListenAndServeKubeletReadOnlyServer initializes a server to respond to HTTP network requests on the Kubelet.
-func ListenAndServeKubeletReadOnlyServer(host HostInterface, resourceAnalyzer stats.ResourceAnalyzer, address net.IP, port uint) {
+func ListenAndServeKubeletReadOnlyServer(
```

Comment:
For now, it may be best to omit tracing on the readonly server, since it is not authenticated.  Ideally, this server would use the WithPublicEndpoint option we removed elsewhere, and the authenticated server would not use the WithPublicEndpoint option.  That way, an unauthenticated user cannot control sampling decisions in the kubelet.

Reply:
Opened https://github.com/kubernetes/kubernetes/issues/109829 to track.

Reply:
sounds good... add a TODO with the reference to that issue here where we construct NewNoopTracerProvider?